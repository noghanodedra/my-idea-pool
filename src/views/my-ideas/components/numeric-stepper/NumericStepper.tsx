import React from 'react';
import NumericInput from 'react-numeric-input';
import styled from 'styled-components';

const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-left: 30px;
`;

const NumericStepper = (props: any) => {
     
    return (
      <StyledContainer>
        <NumericInput
          min={props.min}
          max={props.max}
          value={props.defaultValue}
          onChange={props.onChange}
        />
      </StyledContainer>
    );
}

export default NumericStepper;
