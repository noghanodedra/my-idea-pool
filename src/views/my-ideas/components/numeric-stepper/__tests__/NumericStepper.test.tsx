import 'jest-styled-components';
import 'mutationobserver-shim';

import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';

import NumericStepper from '../NumericStepper';


const renderComponent = () =>
  render(
    <NumericStepper min={1} max={10} defaultValue={8}/>
  );

describe("Numeric Stepper Component", () => {
  beforeEach(() => {
    renderComponent();
  });

  it("renders correctly", async () => {
    expect(
      await screen.getByText(
        (content, element) => element.tagName.toLowerCase() === "span"
      )
    ).toBeDefined();
    expect(
      await screen.getAllByText(
        (content, element) => element.tagName.toLowerCase() === "b"
      )
    ).toHaveLength(2);
  });

  it("is styled correctly", () => {
    const wrapper = shallow(
      <NumericStepper min={1} max={10} defaultValue={8} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("shoule set proper default value", async () => {
    expect(
      await screen.getByText(
        (content, element) => element.tagName.toLowerCase() === "input"
      )
    ).toHaveValue("8");
  });
  
});
