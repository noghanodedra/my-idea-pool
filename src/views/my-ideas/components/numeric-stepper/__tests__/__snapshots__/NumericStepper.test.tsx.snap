// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numeric Stepper Component is styled correctly 1`] = `
<styled.div>
  <NumericInput
    componentClass="input"
    format={null}
    max={10}
    min={1}
    mobile="auto"
    parse={null}
    precision={null}
    step={1}
    strict={false}
    style={Object {}}
    value={8}
  />
</styled.div>
`;
